echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/microsoftteams/MicrosoftTeamsNotificationService.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2020 Cerner Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.echo.microsoftteams;
+
+import com.netflix.spinnaker.echo.api.Notification;
+import com.netflix.spinnaker.echo.controller.EchoResponse;
+import com.netflix.spinnaker.echo.microsoftteams.api.MicrosoftTeamsSection;
+import com.netflix.spinnaker.echo.notification.NotificationService;
+import com.netflix.spinnaker.echo.notification.NotificationTemplateEngine;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+@ConditionalOnProperty("microsoftteams.enabled")
+class MicrosoftTeamsNotificationService implements NotificationService {
+
+  @Autowired MicrosoftTeamsService teamsService;
+
+  @Autowired NotificationTemplateEngine notificationTemplateEngine;
+
+  @Value("${spinnaker.base-url}")
+  String spinnakerUrl;
```

Comment:
Prefer autowired constructors over properties.

Reply:
Converted to autowired constructor